### DISCUSSION TITLE
Simulating V2X communications for CAVs using Waymax

### DISCUSSION THREAD

**Discussion by samanipour** [waymo-research/waymax#84] [Q&A]
As part of my PhD research program, I'm working on Connected and Autonomous Vehicles (CAVs). In addition to simulating autonomous vehicles, I need to be able to simulate Vehicle-to-Vehicle (V2V) and Vehicle-to-Infrastructure(V2I) communications. I want to use the Waymax simulator for my research, but I couldn't find out how to simulate V2X communications in Waymax.

I would appreciate any guidance about simulating V2X communications in Waymax or how we could extend Waymax for such functionality.

**Comment by samuela** [comment]
We don't currently support this, but you could certainly add it yourself by inspecting `SimulatorState`s
